Src/Couchbase.Linq.IntegrationTests/CountQueryTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Couchbase.Linq.Extensions;
+using Couchbase.Linq.IntegrationTests.Documents;
+using NUnit.Framework;
+
+namespace Couchbase.Linq.IntegrationTests
+{
+    [TestFixture]
+    public class CountQueryTests : N1QlTestBase
+    {
+        [Test]
+        public void Count_HasResult()
+        {
+            var context = new CollectionContext(TestSetup.Collection);
+
+            var beers = from beer in context.Query<Beer>()
+                where beer.Type == "beer"
+                select new {beer.Name};
+
+            Console.WriteLine(beers.Count());
+        }
```

Comment:
Shouldn't we be asserting here (and other places)?

Reply:
I'm not sure what to assert on? If the query fails, we'll get an exception. But asserting on the count seems fraught with fragility, one extra document and integration tests fail. Note: these are integration, not unit tests.

Do you have a suggestion on what assertion to add?